src/Windows/R/Components/Impl/InteractiveWorkflow/Implementation/RSessionCallback.cs

```diff
@@ -105,7 +105,7 @@ public Task<PlotDeviceProperties> PlotDeviceCreate(Guid deviceId, CancellationTo
         public async Task ViewLibraryAsync(CancellationToken cancellationToken = default(CancellationToken)) {
             await _coreShell.SwitchToMainThreadAsync(cancellationToken);
             var containerFactory = _coreShell.GetService<IRPackageManagerVisualComponentContainerFactory>();
-            _workflow.Packages.GetOrCreateVisualComponent(containerFactory).Container.Show(focus: true, immediate: false);
+            (_workflow.Packages as IRPackageManagerVisual).GetOrCreateVisualComponent(containerFactory).Container.Show(focus: true, immediate: false);
```

Comment:
I'd like to avoid such downcasting if possible. Since we now have IServiceContainer in place, we can always resolve it as `_services.GetService<IRPackageManagerVisualComponentContainerFactory>().Get(_workflow.Packages)`.
What I would rename though is this factory interface. `IRPackageManagerVisualComponent` to `IRPackageManagerVisual` and `IRPackageManagerVisualComponentContainerFactory` to `IRPackageManagerVisualFactory`.